### PR TITLE
Add new per-framework metrics

### PR DESCRIPTION
**Pull request by philipnrmn**
Fixes [DCOS_OSS-3485](https://jira.mesosphere.com/browse/DCOS_OSS-3485) - Add new mesos framework metrics to mesos_exporter

### CODE REVIEW

**Comment by philipnrmn** [comment]
@gkleiman, mind giving this a look over?

File: master.go [hunk 1]
```diff
@@ -919,7 +920,193 @@ func newMasterCollector(httpClient *httpClient) prometheus.Collector {
...
+				id := matches[1]
+				why := matches[2]
+				c.(*settableCounterVec).Set(value, id, why)
+			}
+			return nil
+		},
+
+		// Per-role resource allocation (min)
+		gauge("master", "frameworks_role_resource_allocation_min", "Number of times resources were filtered per role", "framework_id", "role"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/roles/(.*)/latest_position/min")
```
Comment (gkleiman) [inline master.go]: The right format is:

`master/frameworks/<framework id>.<framework name>/allocation/roles/<role name/latest_position/min`

File: master.go [hunk 2]
```diff
@@ -919,7 +920,193 @@ func newMasterCollector(httpClient *httpClient) prometheus.Collector {
 			return nil
 		},
 
-		// Frameworks metrics
+		// Framework call counts
+		counter("master", "frameworks_calls", "Counts of calls per framework", "framework_id", "call"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/calls/(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
```
[the remark this answers is not in the document]
Comment (philipnrmn) [reply master.go]: yeah we need to update this everywhere. Thanks for the catch.

**Comment by philipnrmn** [comment]
@gkleiman mind approving these changes?

**Comment by gkleiman** [comment]
LGTM, note that the per-framework metrics haven't landed on Mesos yet.

I don't think this PR should be merged until the new metrics are committed to Mesos.

**Comment by philipnrmn** [comment]
Superceded by #74

**Pull request closed without merging**